### PR TITLE
vdk-core: fix error classification of vdk code

### PR DESCRIPTION
**Pull request by tozka**
In case the User code is indirectly responsible (yet closest to the
error in the stack trace), for example using a dependency incorrectly,
the error cause (vdk, or user code) is not recognized correctly.

See more details in https://github.com/vmware/versatile-data-kit/issues/1118 and https://github.com/vmware/versatile-data-kit/issues/580

Testing Done: added functional test covering the scenario

Signed-off-by: Antoni Ivanov <aivanov@vmware.com>

### CODE REVIEW

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py [hunk 1]
```diff
@@ -133,7 +134,21 @@ def invoke_run_function(func: Callable, job_input: IJobInput):
             if arg_name in parameter_names
         }
         if actual_arguments:
-            func(**actual_arguments)
+            try:
+                func(**actual_arguments)
+            except BaseException as e:
```
Comment (doks5) [inline projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:139]: Why not catching `Exception`?

**Review by doks5 — APPROVED** [review]

File: projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py (hunk 1 above)
Comment (tozka) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:139]: To be able to classy properly things lie All Errors - including things like GeneratorExit, MemoryError, etc.

Comment (tozka) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:139]: Actually only GenertorExit,SystemExit and KeyboardInterrupt inherit BaseException (from the built in exceptions). If user code raises them how should we handle them?

Comment (tozka) [reply projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py:139]: users will do whatever they want. 

Here we catch it, pass it through classifier to detect the type of error and then re-raise it wrap in the appropriate VDK exception. 

The question is do we want to handle GenertorExit,SystemExit and KeyboardInterrupt the same way?

**Pull request merged**